src/local.rs

```diff
@@ -182,6 +224,38 @@ impl LocalAsset {
         }
     }
 
+    /// Creates a new .tar.gz file from a provided directory
+    pub fn tar_gz_dir(origin_dir: &str, dest_dir: &str) -> Result<()> {
```

Comment:
would we want a `tar_dir` function that takes a compressionimpl as an arg w/a default instead of having it in the function name?

Reply:
this is what gankra and me were talking about -- our usage of some of the compression enums in `cargo-dist` is bound to how we generate toml files via serde, so i avoided porting those over here (imagine we add a new enum variant to one of them for another project, and then we suddenly introduced new behavior into `cargo-dist` unwillingly)